src/react.ts

```diff
@@ -107,8 +107,13 @@ const createImpl = <T>(createState: StateCreator<T, [], []>) => {
       "[DEPRECATED] Passing a vanilla store will be unsupported in a future version. Instead use `import { useStore } from 'zustand'`.",
     )
   }
-  const api =
+  const api = (
     typeof createState === 'function' ? createStore(createState) : createState
+  ) as WithReact<StoreApi<T>>
+
+  const defaultState = api.getState()
+
+  api.getServerState = () => defaultState
```

Comment:
it doesn't work because `getState()`, even at this point, already returns the state from localstorage. see:
- https://github.com/pmndrs/zustand/pull/2277#issuecomment-1871965063

Reply:
oooops, I missed that comment.